src/OpenTelemetry/Trace/SpanSdk.cs

```diff
@@ -827,7 +828,24 @@ private bool IsAttributeValueTypeSupported(object attributeValue)
                 return true;
             }
 
-            // TODO add array support
+            if (attributeValue is IEnumerable enumerable)
+            {
+                Type entryType = null;
+                foreach (var entry in enumerable)
+                {
+                    if (entryType == null)
+                    {
+                        entryType = entry.GetType();
+                    }
+
+                    if (!this.IsNumericBoolOrString(entry) || entryType != entry.GetType())
+                    {
+                        return false;
```

Comment:
should we just call `ToString()` on all "unsupported" types? This way one can pass an attribute with the Lazy value

Reply:
Yeah, I think that would be useful. I'll separate out the null and type checks. `null` will still set the value to an empty string and unsupported types will use `ToString()`.